src/main/java/com/google/cloud/tools/project/AppYaml.java

```diff
@@ -33,18 +35,13 @@
   private static final String RUNTIME_KEY = "runtime";
 
   /**
-   * @param appyaml A valid existing app.yaml
-   * @throws FileNotFoundException if app.yaml doesn't exist
+   * @param appYaml A valid existing app.yaml
+   * @throws IOException if reading app.yaml fails due to I/O errors
    */
-  public AppYaml(Path appyaml) throws FileNotFoundException {
-    Yaml yaml = new Yaml();
-
-    // Standard snake yaml parsing.
-    @SuppressWarnings("unchecked")
-    Map<String, Object> parserResult =
-        (Map<String, Object>) yaml.load(new FileReader(appyaml.toFile()));
-
-    yamlMap = parserResult;
+  public AppYaml(Path appYaml) throws IOException {
+    try (Reader reader = new FileReader(appYaml.toFile())) {
```

Comment:
FileReader is discouraged because it uses the platform default encoding. Hook an InputStreamReader with a specified encoding to a FileInputStream instead.